### PR TITLE
refactor(date-range-input): Migrate to typescript

### PR DESCRIPTION
**Pull request by Rhotimee**
#### Summary
Migrate date-range-input to typescript

### CODE REVIEW

File: packages/components/inputs/date-range-input/README.md [hunk 1]
```diff
@@ -45,24 +45,53 @@ export default Example;
...
+### Signature `intl`
+
+```ts
+{
+  locale: string;
+  formatMessage: (message: MessageDescriptor) => void;
+}
+```
+
+### Signature `value`
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/date-range-input/README.md:76]: We have the signature explained in the table above. Maybe remove this sub-heading or moving the explanation over here?

Comment (Rhotimee) [reply packages/components/inputs/date-range-input/README.md:76]: This is still related to this [comment](https://github.com/commercetools/ui-kit/pull/2059#discussion_r792494148). 
Btw, the README.md file is automatically generated. 
`yarn generate-readmes`

Comment (emmenko) [inline packages/components/inputs/date-range-input/README.md:76]: Is this still an issue in the generator?

File: packages/components/inputs/date-range-input/src/date-range-input.tsx [hunk 2]
```diff
@@ -411,16 +453,18 @@ class DateRangeCalendar extends Component {
                     onMouseEnter: () => {
                       // we remove the highlight so that the user can use the
                       // arrow keys to move the cursor when hovering
-                      if (isOpen) setHighlightedIndex(null);
+                      if (isOpen) setHighlightedIndex(0);
```
Comment (emmenko) [inline packages/components/inputs/date-range-input/src/date-range-input.tsx:456]: This should probably stay `null`.

File: packages/components/inputs/date-range-input/src/date-range-input.tsx [hunk 3]
```diff
@@ -527,7 +578,7 @@ class DateRangeCalendar extends Component {
                               disabled: this.props.isDisabled,
                               item,
                               onMouseOut: () => {
-                                setHighlightedIndex(null);
+                                setHighlightedIndex(0);
```
Comment (emmenko) [inline packages/components/inputs/date-range-input/src/date-range-input.tsx:581]: Same here

**Review by emmenko — CHANGES_REQUESTED** [review]
Small remark, otherwise looks good.

File: packages/components/inputs/date-range-input/README.md (hunk 1 above)
Comment (Rhotimee) [reply packages/components/inputs/date-range-input/README.md]: No. I needed to rerun the generator script.

**Review by emmenko — APPROVED** [review]

**Pull request merged**